xarray/core/formatting_html.py

```diff
@@ -17,7 +17,16 @@
 
 def short_data_repr_html(array):
     """Format "data" for DataArray and Variable."""
+    from .indexing import PandasIndexAdapter
+    from ..coding.cftimeindex import CFTimeIndex
+
     internal_data = getattr(array, "variable", array)._data
+    # try to convert internal_data.array to CFTimeIndex
+    if isinstance(internal_data, PandasIndexAdapter):
+        if isinstance(internal_data.array, CFTimeIndex):
+            array = array.to_index()
```

Comment:
my motivation is to directly read off the calendar attribute (can already be read off `cftime.DatetimeCALENDAR`) and length (can be already read off `time.size`). 
I was hoping to get the `cftimeindex.__repr__` and the `dataarray.cftimeindex.__repr__` consistent, so `cftimeindex.calendar` would work as well as `dataarray.cftimeindex.calendar`.

First I had implemented `to_index()` without the `if isinstance`'s or `if hasattr`'s. Then I saw the failing tests. This fixed the tests for now and I expected that time/cftime is always 1d array. Is it possible to do multi-dim cftimeindex?

Last motivation: Visually the `cftimeindex.__repr__ looks` nicer than the more plain `numpy_repr`

Reply:
> Is it possible to do multi-dim cftimeindex?

No, pandas Indexes are strictly 1D.

> I was hoping to get the cftimeindex.__repr__ and the dataarray.cftimeindex.__repr__ consistent

We do not do this with other indexes (e.g. underlying Float64Indexes or Int64Indexes), so this would be inconsistent with current behavior.  Furthermore the values in DataArrays are fundamentally stored in arrays, so I think it would be misleading to show an Index object in the DataArray repr.

> so cftimeindex.calendar would work as well as dataarray.cftimeindex.calendar

I agree something along these lines could be nice; however changing the repr will not solve this problem.  Perhaps we can do something using the datetime accessor, e.g. something like:
```python
>>> da.dt.calendar
'noleap'
```
That said attributes on the `dt` accessor normally return DataArrays with the same shape as the original.  Let's maybe move discussion of that to a separate issue.

Reply:
> misleading to show an Index object in the DataArray repr.

agree. I will revert.

> >>> da.dt.calendar
> 'noleap'

I like that idea. I will try to implement this.

cos with only the calendar attribute, as a xr.DataArray coordinate, you need to call:
`ds.time.variable._data.array.calendar`, so `ds.time.dt.calendar` would be more accessible

Reply:
Ok. so this PR will then just implement the CFTimeIndex.calendar and include calendar and length in the repr

Reply:
Sounds good!